mods/default/functions.lua

```diff
@@ -364,10 +356,24 @@ function default.register_fence(name, def)
 	-- Always add to the fence group, even if no group provided
 	def.groups.fence = 1
 
+	local material = def.material
 	def.texture = nil
 	def.material = nil
 
 	minetest.register_node(name, def)
+
+	-- Register crafting recipe, trim away starting colon if any
+	if not material then return end
+	if string.sub(name, 1,1) == ":" then
+		name = string.sub(name, 2)
+	end
```

Comment:
`name:gsub("^:, "")` or similar should do (please double check)

Reply:
Done in 23104f2

Though the if-sub pattern should be faster, I agree that code tidiness is more important than the effectiveness pre-globalstep.

Reply:
Personally I would've preferred `name:match(":?(.*)")` ;)